File: 3760-maximum-substrings-with-distinct-start/3760-maximum-substrings-with-distinct-start.java
class Solution {
    public int maxDistinct(String s) {
        int count = 0;
        int freq[] = new int[26];

        for(char c : s.toCharArray()){
            freq[c-'a']++;
            if(freq[c-'a'] == 1){
                count++;
            }
        }
        return count;
    }
}

File: 3760-maximum-substrings-with-distinct-start/3760-maximum-substrings-with-distinct-start.py
class Solution(object):
    def maxDistinct(self, s):
        """
        :type s: str
        :rtype: int
        """

        freq= [0] * 26
        count = 0
        
        for c in s:
            freq[ord(c) - ord('a')] += 1
            if freq[ord(c) - ord('a')] == 1:
                count += 1

        return count